src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,42 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The logic here removes all animals with health less than or equal to 0 from the `Animal.alive` list. Instead, it should specifically remove only the `herbivore` instance that was bitten. Consider using a condition to filter out only the specific `herbivore` instance.